docs/reference/builder.md

```diff
@@ -1291,6 +1291,74 @@ The `STOPSIGNAL` instruction sets the system call signal that will be sent to th
 This signal can be a valid unsigned number that matches a position in the kernel's syscall table, for instance 9,
 or a signal name in the format SIGNAME, for instance SIGKILL.
 
+## HEALTHCHECK
+
+The `HEALTHCHECK` instruction has two forms:
+
+* `HEALTHCHECK [OPTIONS] CMD command` (check container health by running a command inside the container)
+* `HEALTHCHECK NONE` (disable any healthcheck inherited from the base image)
```

Comment:
Like @tianon said, maybe the first iteration/merge of this feature should probably only have health checking as a status for many reasons.  First we want to make sure it is correct and robust as this is a production feature.  

Second, having actions in docker that are taken on the health check results promotes people to open PRs to docker to add all the functionality that they want.  If heath checks were just a status and we provide a way to look at the last X results, it promotes people, etc to build their logic for acting upon the results outside of Docker.  

Its not that we wouldn't want to code actions into docker, maybe restart policies are the best place to add the exit on failure, but there is no way we can support all of needs of people who want to do something on a health check result.  Having a robust way to see the status, past and present is a better long term solution for this.  

Also the current defaults for retry and exit are very extreme.  A container cannot go from healthy to unhealthy and back to health right now without being killed.  Just because you have a network blip once does not mean that your database should be killed and restarted.  

Changing the defaults to something less extreme would help if we want to support inheritance.